doc/specifications/proposed/content_download/content_download.md

```diff
@@ -0,0 +1,60 @@
+# Content download
+
+> Author: bd@ez.no
+
+## Summary
+This feature covers downloading of Content binary files over HTTP.
+
+It covers:
+- generation of proper links from Twig as well as PHP,
+- permissions checking (the user must be allowed to read the Content)
+- streaming of the file over HTTP
+
+## Components
+
+### Twig helper
+
+> Name: ez_download_path
+> Usage: `ez_download_path( content, 'file' )`
+
+The helper will be used to generate a download link for a given Field/Content.
+
+### Controller action + route
+
+#### Route
+
+Path: `/content/download/{contentId}/{fieldIdentifier}/{fileName}`
+
+Example: ``/content/download/68/file/My file.pdf
+
+Query parameters
+
+- version (optional)
+
+  The version number the file must be downloaded for. Requires the versionview permission.
+  If not specified, the published version is used.
```

Comment:
>  it doesn't apply that much to binary files (it does to images).

video

> What do you think ?

works for proxy, but not for CDN, given several big media customers use eZ Publish/Platform, CDN is a factor to have in mind at least.